plugins/com.google.cloud.tools.eclipse.appengine.libraries/src/com/google/cloud/tools/eclipse/appengine/libraries/Library.java

```diff
@@ -102,4 +104,14 @@ public void setLibraryDependencies(List<String> libraryDependencies) {
       this.libraryDependencies = new ArrayList<>(libraryDependencies);
     }
   }
+
+  public void setRecommendation(LibraryRecommendation recommendation) {
+    if (recommendation != null) {
+      this.recommendation = recommendation;
```

Comment:
True, I totally missed that. Ok, I'll check for null (and in other setter methods as well)